### DISCUSSION TITLE
non-ring atom 0 marked aromatic error.

### DISCUSSION THREAD

**Discussion by etsi123** [rdkit/rdkit#4113] [Q&A] [answered]
Hi everyone,

I have a dataset with some smiles strings which I loop over and get the morgan finger prints for each. A sample code of my method for doing this is below. 

`for smiles in smiles_list: `
.`    mol = Chem.MolFromSmiles(smiles) `
.`    fp = AllChem.GetMorganFingerprintAsBitVector(mol,radius=radius,nBits=nBits)`
.`    fp_arr = np.zeros((1,))`
.`    DataStructs.ConvertToNumpyArray(fp,fp_arr)`
.`   mat.append(fp_arr.tolist())`
`mat = np.array(mat)`

I then use these morgan fingerprints to train a random forest classifier model to predict some property. I'm interested in understanding why certain predictions were made, so I calculate the feature importance of each bit and ultimately construct a list called importantonbits. 

Then I try to draw the morganbits for a specific molecule
`testidx = 100`
`fp,bitinfo = mol2fp(mol_list[100],nBits=nBits,radius=radius)`
`temp = [mols[id],x,bitinfo) for x in importantonbits]`
`Draw.DrawMorganBits(temp) `

where mol2fp is 
`def mol2fp (mol,nbits,radius): `
.`   bitInfo = {}`
.`   fp = AllChem.GetMorganFingerprintAsBitVector(mol,radius=radius,nBits=nBits,bitInfo = bitInfo)`
.`   arr = np.zeros((1,))`
.`    DataStructs.ConvertToNumpyArray(fp,arr)`
. `   return arr,bitInfo`

where I (sometimes) run into this error: RDKit Error: [16:12:25] non-ring atom 0 marked aromatic. How can I fix this? 

I've done some research and it seems to suggest that there may be an error in some (or all?) of the smiles strings possibly, but how on earth can I go back and correct all of them? I wasn't the person who inputted them into the database. Is there a function for cleaning these smiles strings? Also, if there is an error, then why does AllChem.GetMorganFingerprintAsBitVector() work but not Draw.DrawMorganBits()? 

Thank you in advance!

**Comment by greglandrum** [comment, marked as answer]
@etsi123 : it's really difficult for us to provide help about specific things like this if you don't share which version fo the RDKit you are using.
I suspect the problem here is the same as what was reported as #3796. That should be fixed in versions 2020.09.5 and later.

**Reply by etsi123** [reply]
Thank you for providing me with that link, I was able to figure it out with it! Great point, I should have shared the version!
